client/src/partials/cost_center/allocation/allocation.js

```diff
@@ -1,143 +1,149 @@
+  */
+function CostCenterAllocationController ($q, connect, appstate,validate, SessionService) {
+  var vm            = this,
+      dependencies  = {},
+      configuration = vm.configuration = {};
+
+  dependencies.cost_centers = {
+    query : {
+      tables : {
+        'cost_center' : {
+          columns : ['id', 'text', 'note']
         }
       }
-    };
-
-    startup();
-
-    function startup() {
-      $scope.project = SessionService.project;
-      validate.process(dependencies)
-      .then(init);
     }
+  };
+
+  vm.checkAllAvailable  = checkAllAvailable;
+  vm.checkAllAssociated = checkAllAssociated;
+  vm.performChange      = performChange;
+  vm.assign             = assign;
+  vm.remove             = remove;
+  vm.isAssignable       = isAssignable;
+  vm.isRemovable        = isRemovable;
+  vm.state              = 'loading';
+
+  startup();
+
+  function startup() {
+    vm.state = 'loading';
+    vm.project = SessionService.project;
+    validate.process(dependencies)
+    .then(init);
+  }
 
-    function init(models) {
-      angular.extend($scope, models);
-      $scope.acc_1 = { all : false };
-      $scope.acc_2 = { all : false };
-
-      connect.req('/availableAccounts/' + $scope.project.enterprise_id + '/')
-      .then(function (records) {
-        $scope.availableAccounts = records;
-      });
-    }
+  function init(models) {
+    angular.extend(vm, models);
+    vm.acc_1 = { all : false };
+    vm.acc_2 = { all : false };
 
-    function checkAllAvailable() {
-      $scope.availableAccounts.data.forEach(function (item) {
-        item.checked = $scope.acc_1.all;
-      });
-    }
+    connect.req('/availableAccounts/' + vm.project.enterprise_id + '/')
+    .then(function (records) {
+      vm.availableAccounts = records;
+      vm.state = 'loaded';
+    });
+  }
 
-    function checkAllAssociated() {
-      $scope.associatedAccounts.data.forEach(function (item) {
-        item.checked = $scope.acc_2.all;
-      });
-    }
+  function checkAllAvailable() {
+    vm.availableAccounts.data.forEach(function (item) {
+      item.checked = vm.acc_1.all;
+    });
+  }
 
-    function performChange() {
-      $scope.selectedCostCenter = JSON.parse(configuration.costCenter);
-      loadCenterAccount()
-      .then(function (results) {
-        $scope.associatedAccounts = results;
-      });
-    }
+  function checkAllAssociated() {
+    vm.associatedAccounts.data.forEach(function (item) {
+      item.checked = vm.acc_2.all;
+    });
+  }
 
-    function assign() {
-      var accounts = sanitize(); // accounts to associate
-      updateAccounts(accounts)
-      .then(function () {
-        $scope.selectedAccounts.forEach(function (item) {
-          $scope.availableAccounts.remove(item.id);
-          item.checked = false;
-          $scope.associatedAccounts.post(item);
-        });
-      });
-    }
+  function performChange() {
+    vm.selectedCostCenter = JSON.parse(configuration.costCenter);
```

Comment:
It's it's possible for this to fail, there should be a `try { } catch (e) {}` block around the `JSON.parse()` statement.  However, if you are defining `configuration.costCenter` somewhere and are sure that it will always succeed, than this error handling is not necessary.